Dockerfile

```diff
@@ -249,26 +249,28 @@ RUN set -x \
 	&& go build -v -o /usr/local/bin/rsrc github.com/akavel/rsrc \
 	&& rm -rf "$GOPATH"
 
+# Upload docker source
+COPY . /go/src/github.com/docker/docker
+
 # Install runc
 ENV RUNC_COMMIT bbde9c426ff363d813b8722f0744115c13b408b6
 RUN set -x \
-	&& export GOPATH="$(mktemp -d)" \
-  && git clone git://github.com/opencontainers/runc.git "$GOPATH/src/github.com/opencontainers/runc" \
+	&& export GOPATH="/go/src/github.com/docker/docker/tmpsrc" \
+        && mkdir -p $GOPATH \
+        && git clone git://github.com/opencontainers/runc.git "$GOPATH/src/github.com/opencontainers/runc" \
 	&& cd "$GOPATH/src/github.com/opencontainers/runc" \
 	&& git checkout -q "$RUNC_COMMIT" \
-	&& make BUILDTAGS="seccomp apparmor selinux" && make install
+	&& make static BUILDTAGS="seccomp apparmor selinux" && make install
 
 # Install containerd
-ENV CONTAINERD_COMMIT 7146b01a3d7aaa146414cdfb0a6c96cfba5d9091
+ENV CONTAINERD_COMMIT 87846e39081b3b77d2b83748c189f848631e3a48
 RUN set -x \
-	&& export GOPATH="$(mktemp -d)" \
-  && git clone git://github.com/docker/containerd.git "$GOPATH/src/github.com/docker/containerd" \
+	&& export GOPATH="/go/src/github.com/docker/docker/tmpsrc" \
+        && mkdir -p $GOPATH \
+        && git clone git://github.com/docker/containerd.git "$GOPATH/src/github.com/docker/containerd" \
 	&& cd "$GOPATH/src/github.com/docker/containerd" \
 	&& git checkout -q "$CONTAINERD_COMMIT" \
-	&& make && make install
+	&& make static && make install
```

Comment:
should these be static? Not really sure, both containerd and runc are static, but it could easily be a dynamic build. not sure what we want here.